npk/src/npk.rs

```diff
@@ -453,15 +453,15 @@ pub async fn pack(dir: &Path, out: &Path, key: Option<&Path>) -> Result<(), Erro
 pub async fn pack_with(
     dir: &Path,
     out: &Path,
-    key: Option<&Path>,
+    key: Option<PathBuf>,
     squashfs_opts: SquashfsOpts,
 ) -> Result<(), Error> {
     let manifest = read_manifest(dir).await?;
     let name = manifest.name.clone();
     let version = manifest.version.clone();
     let mut builder = Builder::new(dir, manifest);
     if let Some(key) = key {
-        builder = builder.key(key);
+        builder = builder.key(&key);
```

Comment:
The key is passed up to here a s `Option<PathBuf>` just to reference it. This doesn't make sense.

What's wrong with `key: Option<&Path>`?